rs/bitcoin/ckbtc/minter/src/reimbursement/mod.rs

```diff
@@ -0,0 +1,115 @@
+use crate::logs::{P0, P1};
+use crate::memo::MintMemo;
+use crate::state::LedgerMintIndex;
+use crate::{state, CanisterRuntime};
+use candid::{CandidType, Deserialize};
+use ic_canister_log::log;
+use icrc_ledger_types::icrc1::account::Account;
+use icrc_ledger_types::icrc1::transfer::Memo;
+use serde::Serialize;
+
+#[derive(Clone, Eq, PartialEq, Debug, CandidType, Serialize, serde::Deserialize)]
+pub struct ReimburseWithdrawalTask {
+    pub account: Account,
+    /// The amount that should be reimbursed in the smallest denomination.
+    pub amount: u64,
+    pub reason: WithdrawalReimbursementReason,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, CandidType, Serialize, serde::Deserialize)]
+pub struct ReimbursedWithdrawal {
+    pub account: Account,
+    /// The amount that was reimbursed in the smallest denomination.
+    pub amount: u64,
+    pub reason: WithdrawalReimbursementReason,
+    pub mint_block_index: LedgerMintIndex,
+}
+
+pub type ReimbursedWithdrawalResult = Result<ReimbursedWithdrawal, ReimbursedError>;
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum ReimbursedError {
+    /// Whether reimbursement was minted or not is unknown,
+    /// most likely because there was an unexpected panic in the callback.
+    /// The reimbursement request is quarantined to avoid any double minting and
+    /// will not be further processed without manual intervention.
+    Quarantined,
```

Comment:
Nit: Shouldn't we have an error type here? If it's unknown, shouldn't it be something like `UnknownError`?`Quarantined` sounds more like how the error was handled.